From da8218544714e47fbbbdfb1335a936a20eee033b Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Thu, 10 Jan 2013 17:48:45 +0100
Subject: [PATCH 1/1] nl80211: move scan API to wdev

based on linux.wireless commit fd0142844efa85d89017c89227a0f03de1eee327
by Johannes Berg <johannes.berg@intel.com>.

The new P2P Device will have to be able to scan for P2P search, so move
scanning to use struct wireless_dev instead of struct net_device.
---
 src/wl/sys/wl_cfg80211.c |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/src/wl/sys/wl_cfg80211.c b/src/src/wl/sys/wl_cfg80211.c
index c54ae87..6a55728 100644
--- a/src/src/wl/sys/wl_cfg80211.c
+++ b/src/src/wl/sys/wl_cfg80211.c
@@ -46,7 +46,10 @@ u32 wl_dbg_level = WL_DBG_ERR | WL_DBG_INFO;
 
 static s32 wl_cfg80211_change_iface(struct wiphy *wiphy, struct net_device *ndev,
            enum nl80211_iftype type, u32 *flags, struct vif_params *params);
-static s32 wl_cfg80211_scan(struct wiphy *wiphy, struct net_device *ndev,
+static s32 wl_cfg80211_scan(struct wiphy *wiphy,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 6, 0)
+           struct net_device *ndev,
+#endif
            struct cfg80211_scan_request *request);
 static s32 wl_cfg80211_set_wiphy_params(struct wiphy *wiphy, u32 changed);
 static s32 wl_cfg80211_join_ibss(struct wiphy *wiphy, struct net_device *dev,
@@ -482,9 +485,15 @@ wl_cfg80211_change_iface(struct wiphy *wiphy, struct net_device *ndev,
 }
 
 static s32
-wl_cfg80211_scan(struct wiphy *wiphy, struct net_device *ndev,
+wl_cfg80211_scan(struct wiphy *wiphy,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 6, 0)
+                 struct net_device *ndev,
+#endif
                  struct cfg80211_scan_request *request)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0)
+	struct net_device *ndev = request->wdev->netdev;
+#endif
 	struct wl_cfg80211_priv *wl = ndev_to_wl(ndev);
 	struct cfg80211_ssid *ssids;
 	struct wl_cfg80211_scan_req *sr = wl_to_sr(wl);
-- 
1.7.9.5

